src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -147,29 +150,31 @@ private static void initHttpClient() {
 
     // proxy settings
     if ("true".equalsIgnoreCase(System.getProperty(USE_PROXY))) {
-      if (System.getProperty(PROXY_PORT) == null) {
-        throw new IllegalArgumentException(
-            "proxy port number is not provided, please assign proxy port to http.proxyPort option");
-      }
-      if (System.getProperty(PROXY_HOST) == null) {
-        throw new IllegalArgumentException(
-            "proxy host IP is not provided, please assign proxy host IP to http.proxyHost option");
-      }
-      String proxyHost = System.getProperty(PROXY_HOST);
-      int proxyPort = Integer.parseInt(System.getProperty(PROXY_PORT));
-      HttpHost proxy = new HttpHost(proxyHost, proxyPort, PROXY_SCHEME);
-      DefaultProxyRoutePlanner routePlanner = new DefaultProxyRoutePlanner(proxy);
-      clientBuilder = clientBuilder.setRoutePlanner(routePlanner);
-
-      // Check if proxy username and password are set
-      final String proxyUser = System.getProperty(HTTP_PROXY_USER);
-      final String proxyPassword = System.getProperty(HTTP_PROXY_PASSWORD);
-      if (!isNullOrEmpty(proxyUser) && !isNullOrEmpty(proxyPassword)) {
-        Credentials credentials = new UsernamePasswordCredentials(proxyUser, proxyPassword);
-        AuthScope authScope = new AuthScope(proxyHost, proxyPort);
-        CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
-        credentialsProvider.setCredentials(authScope, credentials);
-        clientBuilder = clientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+      if(!bypassProxy(account)) {
```

Comment:
nit: The two if checks could be combined into one
```suggestion
      if("true".equalsIgnoreCase(System.getProperty(USE_PROXY)) && !bypassProxy(account)) {
```